src/components/Icons/index.js

```diff
@@ -0,0 +1,56 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import config from 'vtex-tachyons/config.json'
+
+export default class Icon extends Component {
+  constructor() {
+    super()
+
+    this.state = {
+      module: null,
+    }
+  }
+
+  formatHexValue(value) {
+    const HEX_COLOR = /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/
+    const formatedValue = value.indexOf('#') === -1 ? `#${value}` : value
+
+    return HEX_COLOR.test(formatedValue)
+      ? formatedValue
+      : config.colors['serious-black']
+  }
+
+  toTitleCase(str) {
+    return str.replace(/\w\S*/g, function(txt) {
+      return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
+    })
+  }
+
+  componentDidMount() {
+    const { type } = this.props
+
+    import(`./${this.toTitleCase(type)}`).then(module =>
+      this.setState({ module: module.default })
```

Comment:
Verificar se componente ainda esta montado ainda de chamar esse `setState` para não dar o erro de "`setState` is not defined"

Reply:
Esse `import` está dentro do `componentDidMount`.

Reply:
O `then` é executado assincronamente, se o componente desmontar antes do `then` ser executado, ao entrar no `then` e chamar `setState` vai dar um erro, pq o componente já nao existe mais.